config/main.py

```diff
@@ -519,7 +519,99 @@ def is_ipaddress(val):
+            # expire call.
+            if self.client.hsetnx(self.lockName, "PID", self.pid):
+                self.client.expire(self.lockName, self.timeout)
+            # if lock exists but expire timer not running, run expire time and
+            # abort.
+            elif not self.client.ttl(self.lockName):
+                click.echo(":::Unable to acquire lock. Resetting timer and aborting:::");
+                self.client.expire(self.lockName, self.timeout)
+                sys.exit(1)
+            else:
+                click.echo(":::Unable to acquire lock. Aborting:::");
+                sys.exit(1)
+        except Exception as e:
+            click.echo(":::Exception: {}:::".format(e))
+            sys.exit(1)
+        return
 
+    def reacquireLock(self):
+        try:
+            # Try to set lock first
+            if self.client.hsetnx(self.lockName, "PID", self.pid):
+                self.client.expire(self.lockName, self.timeout)
+            # if lock exists, check who owns it
+            else:
+                p = self.client.pipeline(True)
+                # watch, we do not want to work on modified lock
+                p.watch(self.lockName)
+                # if current process holding then extend the timer
+                if p.hget(self.lockName, "PID") == str(self.pid):
+                    self.client.expire(self.lockName, self.timeout)
+                    p.unwatch()
+                    return
+                else:
+                    # some other process is holding the lock.
+                    click.echo(":::Unable to reacquire lock (lock PID: {}, self.pid: {}):::".\
+                        format(p.hget(self.lockName, "PID"), self.pid))
+                    p.unwatch()
+                    sys.exit(1)
+
+        except Exception as e:
+            click.echo(":::Exception: {}:::".format(e))
+            sys.exit(1)
+        return
+
+    def _releaseLock(self):
+        try:
+            p = self.client.pipeline(True)
+            # watch, we do not want to work on modified lock
+            p.watch(self.lockName)
+            # if current process holding the lock then release it.
+            if p.hget(self.lockName, "PID") == str(self.pid):
+                p.multi()
+                p.delete(self.lockName)
+                p.execute()
+                return
+            # lock may be None, if timer has expired before releasing lock.
+            elif not self.lockName:
+                return
+            else:
+                # some other process is holding the lock.
+                click.echo(":::Unable to release lock (lock PID: {}, self.pid: {}):::".\
+                    format(p.hget(self.lockName, "PID"), self.pid))
+            p.unwatch()
+        except Exception as e:
+            click.echo(":::Exception: {}:::".format(e))
+        return
+
+    def __del__(self):
+        self._releaseLock()
+        return
+# end of class configdblock
+
+cdblock = ConfigDbLock()
```

Comment:
Can you also rename this variable to make it more readable? "cdblock" can be mistaken for "CD Block". Maybe `configdb_lock`?

Reply:
I think, let`s use\force camel case for variables in this file. I did that already inside the class, so If you do not prefer otherwise I will use below var name.
 
```
configDbLock = ConfigDbLock()
```

Reply:
For our style in SONiC, we tend to use Pascal case for class names and lower_with_underscores for variable names.

Reply:
@jleveque , cool thx for sharing, I will stick to it then.